[Concept Entry] Java / Boolean Logic (#4225)
Codecademy/docs: content/java/concepts/boolean-logic/boolean-logic.md

---
Title: 'Boolean Logic'
Description: 'A set of operators and methods in Java for manipulating booleans.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Booleans'
  - 'Logic'
  - 'Logical'
  - 'Operators'
  - 'Conditionals'
  - 'Java'
CatalogContent:
  - 'learn-java'
  - 'paths/introduction-to-android-with-java'
---

One of the primitive [data types](https://www.codecademy.com/resources/docs/java/data-types) in Java is the `boolean`. A boolean object takes a value of `true` or `false`.

> **Boolean logic** describes the ways in which boolean values can be combined and manipulated. Java implements boolean logic through a set of [operators](https://www.codecademy.com/resources/docs/java/operators) and [methods](https://www.codecademy.com/resources/docs/java/methods). These are described below.

Reviewer: ```suggestion
**Boolean logic** describes how boolean values can be combined and manipulated. Java implements boolean logic through a set of [operators](https://www.codecademy.com/resources/docs/java/operators) and [methods](https://www.codecademy.com/resources/docs/java/methods).
```

|-|-|-|-|
| `&&` | `(true) && (false)` | Returns `true` if both arguments equal `true`; otherwise returns `false` | AND / Conjunction |
| `\|\|` | `(true) \|\| (false)` | Returns `true` if at least one argument equals `true`; otherwise returns `false` | OR / Inclusive disjunction |
| `!` | `!(true)` | Returns `true` if the argument equals `false`; otherwise returns `false` | NOT / Negation |
| `^` | `(true) ^ (false)` | Returns `true` if exactly one argument equals `true`; otherwise returns `false` | XOR / Exclusive disjunction |

## Boolean Logical Methods

The boolean logical methods available in Java are described in the following table:

| Method Syntax | Behavior | Equivalent Logical Concept |
|-|-|-|
| `logicalAnd(boolean A, boolean B)` | Returns the result of `(A && B)` | AND / Conjunction |
| `logicalOr(boolean A, boolean B)` | Returns the result of `(A \|\| B)` | OR / Inclusive disjunction |
| `logicalXor(boolean A, boolean B)` | Returns the result of `(A ^ B)` | XOR / Exculsive disjunction |

## Use in Conditionals

Boolean logic is commonly used within conditionals to control the flow of a program.

> ### Codebyte Example

Reviewer: Lets make this is a simple example section. As our online compiler doesn't support java.

---
Title: 'Boolean Logic'
Description: 'A set of operators and methods in Java for manipulating booleans.'
Subjects:

>   - 'Computer Science'

Reviewer 2: ```suggestion
  - 'Computer Science'
  - 'Code Foundations'
```
